proto/sentry_protos/sentry/v1alpha/taskworker.proto

```diff
@@ -0,0 +1,119 @@
+syntax = "proto3";
+
+package sentry_protos.sentry.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+
+message RetryState {
+  // Current attempt number
+  int32 attempts = 1;
+
+  // The classname or adapter type for the retry policy
+  string kind = 2;
+
+  // After this attempt the task should be discarded
+  optional int32 discard_after_attempt = 3;
+
+  // After this attempt the task should be put in the dead-letter-queue.
+  optional int32 deadletter_after_attempt = 4;
+}
+
+enum TaskActivationStatus {
+  TASK_ACTIVATION_STATUS_UNSPECIFIED = 0;
+  TASK_ACTIVATION_STATUS_PENDING = 1;
+  TASK_ACTIVATION_STATUS_PROCESSING = 2;
+  TASK_ACTIVATION_STATUS_FAILURE = 3;
+  TASK_ACTIVATION_STATUS_RETRY = 4;
+  TASK_ACTIVATION_STATUS_COMPLETE = 5;
+}
+
+// Task message that is stored in Kafka.
+// Once consumed, TaskActivations are wrapped with InflightActivation to track additional state
+message TaskActivation {
+  // A GUID for the task. Used to update tasks
+  string id = 1;
+
+  // The name of the task. This name is resolved within the worker
+  string taskname = 2;
+
+  // An opaque parameter collection. Could be JSON or protobuf encoded
+  string parameters = 3;
+
+  // A map of headers for the task.
+  map<string, string> headers = 4;
+
+  // The timestamp a task was stored in Kafka
+  google.protobuf.Timestamp received_at = 5;
+
+  // The task deadline. After this time the task is expired and should not be executed.
+  optional google.protobuf.Timestamp deadline = 6;
+
+  // Retry state
+  RetryState retry_state = 7;
+}
+
+// Once a TaskActivation has been received by the task consumer it is wrapped
+// with InflightActivation so that processing state can be tracked.
+// This proto might not be used as InflightActivations don't need to cross process boundaries.
+message InflightActivation {
```

Comment:
> How are we generating this id and how do we ensure it is unique for each datastore?

I was thinking we could use the `TaskActivation.id` for the `InflightActivation` identifier. Currently task activations use `uuid4()` to generate their id values.

If we're happy with that approach, I can update the django model to have a `UUID` primary key to keep the prototype working.

Reply:
Sounds good, that schema implementation makes sense. Using `uuid4()` also seems ok to me. An alternative might be to hash some metadata of the task (topic, partition, offset) to avoid collisions. However, this implementation detail is probably outside the scope of this PR. Regardless `TaskActivation.id` will be a string.